File: scripts/sqlite-query.py
#!/usr/bin/env python3
"""Helper: reads SQL from stdin, queries SQLite DB, outputs JSON to stdout."""
import sqlite3, json, sys

if len(sys.argv) < 2:
    print("[]")
    sys.exit(0)

db_path = sys.argv[1]
sql = sys.stdin.read().strip()
if not sql:
    print("[]")
    sys.exit(0)

try:
    db = sqlite3.connect(db_path)
    db.row_factory = sqlite3.Row
    cursor = db.cursor()
    cursor.execute(sql)
    rows = [dict(r) for r in cursor.fetchall()]
    print(json.dumps(rows, ensure_ascii=False))
    db.close()
except Exception:
    print("[]")

File: scripts/sqlite-query.js
#!/usr/bin/env node
// Helper script: reads SQL from stdin, queries SQLite DB, outputs JSON to stdout
const Database = require('better-sqlite3');
const fs = require('fs');

const dbPath = process.argv[2];
if (!dbPath || !fs.existsSync(dbPath)) {
  console.log('[]');
  process.exit(0);
}

// Read SQL from stdin
let sql = '';
try {
  sql = fs.readFileSync(0, 'utf-8').trim();
} catch {
  console.log('[]');
  process.exit(0);
}

if (!sql) {
  console.log('[]');
  process.exit(0);
}

try {
  const db = new Database(dbPath, { readonly: true });
  const rows = db.prepare(sql).all();
  console.log(JSON.stringify(rows));
  db.close();
} catch (e) {
  console.log('[]');
}
